src/commands/start.ts

```diff
@@ -41,11 +46,44 @@ export async function handler(
 
   const app = await createServer(config.port, config);
   debug('Start server on port %d', config.port);
-  return new Promise(resolve => {
-    app.listen(config.port, async () => {
+  return new Promise((resolve, reject) => {
+    const serverStartedSuccessfully = async () => {
       printRouteInfo(config);
       resolve();
-    });
+    };
+    const handleServerError = async (error: ServerError) => {
+      if (error.code === 'EADDRINUSE') {
+        const answers = await inquirer.prompt([
+          {
+            type: 'input',
+            default: config.port + 1,
+            name: 'newPortNumber',
+            message: `Port ${config.port} is already in use. Choose a new port number:`,
+            validate: input => {
+              const newPortNumber = parseInt(input, 10);
+              if (
+                !Number.isNaN(newPortNumber) &&
+                newPortNumber <= 65535 &&
+                newPortNumber > 0
+              ) {
+                return true;
+              }
+              return 'Please enter a port number between 0 and 65535.';
+            },
+          },
+        ]);
+        const server = app.listen(
+          answers.newPortNumber,
+          serverStartedSuccessfully
+        );
+        server.on('error', handleServerError);
+      } else {
+        reject(error);
+      }
+    };
```

Comment:
Should we care about placing a max limit on the number of times to prompt a user for an available port?

Reply:
Good point, I've added a maximum of 3 retries then exit with an error message. Thanks.